runtime/lua/vim/lsp/util.lua

```diff
@@ -1892,14 +1892,19 @@ function M.symbols_to_items(symbols, bufnr, position_encoding)
       local end_lnum = range['end'].line + 1
       local end_col = get_line_byte_from_position(bufnr, range['end'], position_encoding) + 1
 
+      local is_deprecated = symbol.deprecated
+        or (symbol.tags and vim.tbl_contains(symbol.tags, protocol.SymbolTag.Deprecated))
+      local text =
+        string.format('[%s] %s%s', kind, symbol.name, is_deprecated and ' (deprecated)' or '')
```

Comment:
Would be nice to have a strike-through and/or a highlight but I guess this is good enough for the quickfix/location list.